src/cli/commands/import.js

```diff
@@ -276,12 +307,45 @@ export class Import extends Install {
     this.resolver = new ImportPackageResolver(this.config, this.lockfile);
     this.linker = new PackageLinker(config, this.resolver);
   }
-
+  createLogicalDependencyTree(packageJson: ?string, packageLock: ?string): LogicalDependencyTree {
+    try {
+      if (!packageJson || !packageLock) {
+        throw new Error('files missing');
```

Comment:
this string should probably come from `reporter.lang` and maybe report what files are expected.

Actually, since their existence is already checked around line 337-ish, do these even need to be 'nullable' types? Although I imagine Flow complains if you change `?string` to just `string`

Reply:
Yeah, this is some `flow` acrobatics. I don't like it either.
I thought not to bother with `reporter.lang` in this case because it will be caught by the below try/catch block by definition. I'll try to work out a better way.

Reply:
Change `createLogicalDependencyTree` to only accept non-nullable types, then use `invariant` calls before calling it at line  340 to instruct flow that they cannot be null. Something like this:

```
invariant(packageJson, 'packageJson should exist');
invariant(packageLock, 'packageJson should exist');
```

That being said, the condition at line 336 should be enough - maybe the fact that you're storing this inside a variable is throwing Flow off. Maybe you can try to split it?

```
let importSource = 'node_modules';
if (packageJson && packageLock && semver.satisfies(nodeVersion, '>=5.0.0')) {
    importSource = 'package-lock.js';
}
```